src/extension/app/store/app.js

```diff
@@ -532,7 +532,22 @@ export class AppStore {
    */
   getContentSourceLabel() {
     const { preview } = this.status;
-    return preview?.sourceLocation.includes('onedrive:') ? 'SharePoint' : 'Google Drive';
+    const { sourceLocation } = preview;
+
+    if (sourceLocation) {
+      return sourceLocation.includes('onedrive:')
+        ? 'SharePoint'
+        : sourceLocation.includes('gdrive:')
+          ? 'Google Drive'
+          : 'BYOM';
```

Comment:
Still not convinced we need to explicitly check for BYOM origins and this fallback is enough. If you feel strongly on it I can add it ;)

Reply:
For added safety, I would switch to `startsWith()`, and I  think we can explicitly check for `onedrive:`, `gdrive:` or `markup:` here. There is no other possible source location prefix.